onnxruntime/core/providers/cpu/rnn/deep_cpu_lstm.cc

```diff
@@ -794,17 +794,17 @@ void UniDirectionalLstm<T>::Compute(const gsl::span<const T>& inputs_arg,
   // explicitly check just the range for each iteration, however if it's going to run over
   // it should also run over on the last iteration, so this should be good enough to catch any
   // logic errors causing bounds violations.
-  span_T_iter C_prev_end = batched_internal_state_prev_one_step.end();
-  span_T_iter C_prev_clipped_end = batched_internal_state_clipped_one_step.end();
-  span_T_const_iter previous_state_end = batched_hidden_state_one_step.end();
-
   if (batch_parallel_) {
```

Comment:
This comment was attached to the things you moved. Do C_prev_end and C_prev_clipped_end need to move? Looked like only previous_state_end was the issue.

Reply:
I will move the comment. C_prev_end and the other one are passed and updated by non-const ref in GateComputations()

---
In reply to: [327850516](https://github.com/microsoft/onnxruntime/pull/1905#discussion_r327850516) [](ancestors = 327850516)

Reply:
They can be made const in the call to GateComputations I believe. They've just used for bounds checking and don't change across iterations as per the comment.

---
In reply to: [327855146](https://github.com/microsoft/onnxruntime/pull/1905#discussion_r327855146) [](ancestors = 327855146,327850516)